pkg/loop/internal/pb/relayer.proto

```diff
@@ -0,0 +1,262 @@
+  rpc Sign (SignRequest) returns (SignReply) {}
+}
+
+message KeysReply {
+  repeated string accounts = 1;
+}
+
+message SignRequest {
+  string account = 1;
+  bytes data = 2;
+}
+
+message SignReply {
+  bytes signed = 1;
+}
+
+service Relayer {
+  rpc NewConfigProvider (NewConfigProviderRequest) returns (NewConfigProviderReply) {}
+  rpc NewMedianProvider (NewMedianProviderRequest) returns (NewMedianProviderReply) {}
+  rpc NewMercuryProvider (NewMercuryProviderRequest) returns (NewMercuryProviderReply) {}
+
+  rpc ChainStatus (ChainStatusRequest) returns (ChainStatusReply) {}
+  rpc ChainStatuses (ChainStatusesRequest) returns (ChainStatusesReply) {}
+
+  rpc NodeStatuses (NodeStatusesRequest) returns (NodeStatusesReply) {}
+
+  rpc SendTx (SendTxRequest) returns (google.protobuf.Empty) {}
+}
+
+message RelayArgs {
+  bytes externalJobID = 1; // [32]byte
+  int32 jobID = 2;
+  string contractID = 3;
+  bool new = 4;
+  bytes relayConfig = 5;
+}
+
+message PluginArgs {
+  string transmitterID = 1;
+  bytes pluginConfig = 2;
+}
+
+message NewConfigProviderRequest {
+  RelayArgs relayArgs = 1;
+}
+
+message NewConfigProviderReply {
+  uint32 configProviderID = 1;
+}
+
+message NewMedianProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+message NewMedianProviderReply {
+  uint32 medianProviderID = 1;
+}
+
+message NewMercuryProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+message NewMercuryProviderReply {
+  uint32 mercuryProviderID = 1;
+}
+
+message ChainStatusRequest {
+  string id = 1;
+}
+
+message ChainStatusReply {
+  ChainStatus chain = 1;
+}
+
+message ChainStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+}
+
+message ChainStatusesReply {
+  repeated ChainStatus chains = 1;
+  int32 count = 2;
+}
+
+message ChainStatus {
+  string id = 1;
+  bool enabled = 2;
+  string config = 3; // TOML
+}
+
+message NodeStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+  repeated string chainIDs = 3;
+}
+
+message NodeStatusesReply {
+  repeated NodeStatus nodes = 1;
+  int32 count = 2;
+}
+
+message NodeStatus {
+  string chainID = 1;
+  string name = 2;
+  string config = 3; // TOML
+  string state = 4;
+}
+
+message SendTxRequest {
+  string chainID = 1;
+  string from = 2;
+  string to = 3;
+  bytes amount = 4; // big.Int
+  bool balanceCheck = 5;
+}
+
+service DataSource {
+  rpc Observe (google.protobuf.Empty) returns (ObserveReply) {}
+}
+
+message ObserveReply {
+  bytes value = 1; // big.Int
+}
+
+service OffchainConfigDigester {
+  rpc ConfigDigest (ConfigDigestRequest) returns (ConfigDigestReply) {}
+  rpc ConfigDigestPrefix (ConfigDigestPrefixRequest) returns (ConfigDigestPrefixReply) {}
+}
+
+message ContractConfig {
+  bytes configDigest = 1; // [32]byte
+  uint64 configCount = 2;
+  repeated bytes signers = 3; // []OnchainPublicKey
+  repeated string transmitters = 4; // []Account
+  uint32 F = 5; // uint8
```

Comment:
https://github.com/smartcontractkit/libocr/blob/master/offchainreporting2/types/types.go#L103-L104

Reply:
can we tie/link this and the other proto definitions to the libocr code? it would be great to indicate in general that this is all or mostly a wrapper to libocr and point the reader/maintainer to that code base for more documentation and context

Reply:
Updated all of the messages to point to the corresponding types/methods.